### PR TITLE
:robot: Python version --- Change to 3.11

### PR DESCRIPTION
**Pull request by jameshughes89**
### What

Update to python 3.11

### Testing

:+1:

### Additional Notes
Idk if I really wanna' merge yet. However, I do want to see if the actions work.

### CODE REVIEW

File: .github/workflows/sphinx-build.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@v4
       with:
-        python-version: 3.10.0
```
Comment (twentylemon) [inline .github/workflows/sphinx-build.yml:20]: Any reason to force specify `3.11.0`? `3.11` seems specific enough.

Comment (jameshughes89) [reply .github/workflows/sphinx-build.yml:20]: Legit, because it was 3.10.0, and before it was 3.9.0. 

From the docs, having only 3.x would be "faster".

Comment (jameshughes89) [reply .github/workflows/sphinx-build.yml:20]: I used bad syntax there. 

I meant 3.SOMETHING without the 3rd part of the version number.

Comment (twentylemon) [reply .github/workflows/sphinx-build.yml:20]: Gotcha. That is how duckbot specifies [3.8](https://github.com/duck-dynasty/duckbot/blob/main/.github/actions/setup-python-venv/action.yml#L26). It is the way to go, imo.

**Comment by jameshughes89** [comment]
It's just 3.11 now. 

I'll make sure it's happy on windows before I merge. Other repos will be updated once this one is all good.

**Comment by jameshughes89** [comment]
Removed the legacy thing for the windows venv setup description since it's not needed. 

Everything was fine on windows after updating locally. PyCharm seems to be convinced that it's still using 3.10 though, but it's defo not... 🤷 

image
image
image
image

**Comment by jameshughes89** [comment]
> Removed the legacy thing for the windows venv setup description since it's not needed. 
> 
> Everything was fine on windows after updating locally. PyCharm seems to be convinced that it's still using 3.10 though, but it's defo not... 🤷 
> 
> image
> image
> image
> image
> 

I found people talking about a similar issue with pycharm showing the wrong python version online. Some said it resolved after a pycharm update. I'll see what happens.

**Review by twentylemon — APPROVED** [review]

**Pull request merged**